src/main/java/co/cask/cdap/guides/workflow/ProblemLogger.java

```diff
@@ -0,0 +1,18 @@
+package co.cask.cdap.guides.workflow;
+
+import co.cask.cdap.api.workflow.AbstractWorkflowAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a dummy action in the Workflow that could be used to send email notifications.
+ */
+public class ProblemLogger extends AbstractWorkflowAction {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ProblemLogger.class);
+
+  @Override
+  public void run() {
+    LOG.info("Send email notification here about bad records.");
```

Comment:
Could this be modified to show receiving the context, and thus the passing of information from the Workflow to the ProblemLogger?